dask/array/tests/test_array_utils.py

```diff
@@ -87,3 +89,29 @@ def test_meta_from_array_type_inputs():
 def test_assert_eq_checks_dtype(a, b):
     with pytest.raises(AssertionError, match="a and b have different dtypes"):
         assert_eq(a, b)
+
+
+@pytest.mark.parametrize(
+    "a,b",
+    [
+        (1.0, 1.0),
+        ([1, 2], [1, 2]),
+        (da.array([1, 2]), da.array([1, 2])),
+    ],
+)
+def test_assert_eq_scheduler(a, b):
+    counter = 0
+
+    def custom_scheduler(*args, **kwargs):
+        nonlocal counter
+        counter += 1
+        return get_sync(*args, **kwargs)
+
+    assert_eq(a, b)
+    assert counter == 0
+
+    assert_eq(a, b, scheduler=custom_scheduler)
+    if isinstance(a, Array) and isinstance(b, Array):
+        assert counter == 2
+    else:
+        assert counter == 0
```

Comment:
I really like this test! But this part took me a minute to understand. I am wondering if something like this is clearer:

```suggestion
    n_da_arrays = len([x for x in [a, b] if isinstance(x, Array)])
    assert counter == n_da_arrays
```

Reply:
That's so smart! Committed, thank you.